### PR TITLE
Remove compilers dependency from conda environment and force install x64 git in OSX Github Action workflows

### LINKED ISSUE #2419: Remove `compilers` from `environment.yml`
See: https://github.com/conda-forge/esmvalcore-feedstock/pull/74#discussion_r1600090885

### PR DESCRIPTION
**Pull request by valeriupredoi**
## Description

Closes #2419 

`compilers` was added by @bouweandela four years ago in #603 - see https://github.com/ESMValGroup/ESMValCore/pull/603#issuecomment-631651179 - I don't think we still need it but may be worth checking the docker build goes well without it anyway

OSX tests: failing with "setuptools_scm._run_cmd.CommandNotFoundError: git" - this has been seen in https://github.com/ESMValGroup/ESMValTool/pull/3581 and it's not hitting just this node, but rather, it's an endemic issue related to macOS 14 on arm64:
```
  Image: macos-14-arm64
  Version: 20240422.3
  Included Software: https://github.com/actions/runner-images/blob/macos-14-arm64/20240422.3/images/macos/macos-14-arm64-Readme.md
  Image Release: https://github.com/actions/runner-images/releases/tag/macos-14-arm64%2F20240422.3
```
For some reason, the system git goes fishing when the x64 environment is activated and used - @schlunma and I had a decent debate about it in the linked Tool PR - no time to really-really understand the issue, just plop a git installer in the workflow!

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ on:
   push:
     branches:
       - main
+      - remove_compilers_dependency
```
Comment (bouweandela) [inline .github/workflows/run-tests.yml:24]: Clean up?

**Review by bouweandela — COMMENTED** [review]
Looks fine to me @valeriupredoi, could you clean up a bit?

The compilers package was needed a long time ago, when we were still installing most packages from PyPI and only those that weren't on PyPI from conda.

File: .github/workflows/run-tests.yml (hunk 1 above)
Comment (valeriupredoi) [reply .github/workflows/run-tests.yml:24]: yessir, sorry

**Review by bouweandela — APPROVED** [review]

**Comment by valeriupredoi** [comment]
Thanky, bud 🍺

**Pull request merged**